r/R/dataset.R

```diff
@@ -93,8 +93,19 @@ open_dataset <- function(sources,
     return(dataset___UnionDataset__create(sources, schema))
   }
   factory <- DatasetFactory$create(sources, partitioning = partitioning, ...)
-  # Default is _not_ to inspect/unify schemas
-  factory$Finish(schema, isTRUE(unify_schemas))
+  
+  tryCatch(
+    # Default is _not_ to inspect/unify schemas
+    factory$Finish(schema, isTRUE(unify_schemas)),
+    error = function (e) {
+      msg <- conditionMessage(e)
+      if(grep("Parquet magic bytes not found in footer", msg)){
+        stop("Looks like these are not parquet files, did you mean to specify a 'format'?", call. = FALSE)
+      } else {
+        stop(e)
```

Comment:
Does this `stop()` return the call trace now?

Reply:
I think so? When I click "Show Traceback" in RStudio, I get this;

```
> ds <- open_dataset(csv_dir, partitioning = "part")
 Error: Looks like these are not parquet files, did you mean to specify a 'format'? 
6. stop("Looks like these are not parquet files, did you mean to specify a 'format'?", 
    call. = FALSE) at dataset.R#103
5. value[[3L]](cond) 
4. tryCatchOne(expr, names, parentenv, handlers[[1L]]) 
3. tryCatchList(expr, classes, parentenv, handlers) 
2. tryCatch(factory$Finish(schema, isTRUE(unify_schemas)), error = function(e) {
    msg <- conditionMessage(e)
    if (grep("Parquet magic bytes not found in footer", msg)) {
        stop("Looks like these are not parquet files, did you mean to specify a 'format'?",  ... at dataset.R#97
1. open_dataset(csv_dir, partitioning = "part") 
```

Reply:
Ah, I was thinking about the line 105 stop. I'm not certain if passing `e` there will have the same behavior as it would now (for errors that aren't this parquet issue, like providing a non-existent directory). Currently (on the master branch) that looks like:

```
> ds <- open_dataset("~/does/not/exist") 
Error: IOError: Cannot list directory '/Users/jkeane/does/not/exist'. Detail: [errno 2] No such file or directory
```

But with this change it might be something like the following without also adding `call. = FALSE` here too (though I'm not certain of that!)

```
Error in open_dataset() : Error: IOError: Cannot list directory '/Users/jkeane/does/not/exist'. Detail: [errno 2] No such file or directory
```

Reply:
Now I've updated the error to match the patterns mentioned by @nealrichardson below, so I think that resolves this.